### PR TITLE
Bump cardano-node to 1.34.1

### PR DESCRIPTION
**Pull request by Anviking**
- [x] Bump node to 1.34.1
    - [x] Bump node
    - [x] Tweak readme
    - [x] Fix a few new compilation warnings/errors
- [x] Drop support for `stack build`
    - Tweak docs
    - Remove `stack build` step from buildkite CI

### Comments

<s>Blocked by `stack` + sublib problem</s>

Guided by `git diff 1.33.0 1.34.0 -- cabal.project` in `cardano-node`

### Issue Number

ADP-1433

### CODE REVIEW

**Review by jonathanknowles — APPROVED** [review]
Looks very reasonable to me. 👍🏻 

I'm guessing that all of the `nix/materialized/stack-nix/.stack-to-nix.cache.*` files were updated automatically, via `nix/regenerate.sh`?

Should the title of this PR be "Bump cardano-node to 1.34.1", given the changes within the `README.md`?

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

File: docs/developers/Building.md [hunk 1]
```diff
@@ -6,28 +6,15 @@
...
-```
-stack build --test --no-run-tests
-```
-
-You may need to install the [`libsodium-dev`](https://doc.libsodium.org/installation), `libghc-hsopenssl-dev`, `gmp`, `sqlite` and `systemd` development
-libraries for the build to succeed.
-
 ## Cabal
 
 Alternatively, it's possible to build this project with [Cabal][].
```
Comment (Anviking) [inline docs/developers/Building.md]:
```suggestion
```

**Review by sevanspowell — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**